pkg/util/hlc/hlc.go

```diff
@@ -125,6 +134,67 @@ func NewClock(physicalClock func() int64, maxOffset time.Duration) *Clock {
 	}
 }
 
+// StartMonitoringClockJumps starts a goroutine to update the clock's
+// maxClockJump based on the values pushed in maxClockJumpCh.
+//
+// This also keeps lastPhysicalTime up to date to avoid spurious jump errors.
+// maxClockJumpCh is a channel from which the tolerated clock physical jump
+// threshold is read.
+//
+// A nil channel or a value of 0 for maxClockJump disables checking clock
+// jumps between two successive reads of the physical clock.
+//
+// This should only be called once per clock, and will return an error if called
+// more than once
+//
+// tickerFn is used to create a new ticker
+//
+// tickCallback is called whenever maxClockJumpCh or a ticker tick is processed
+func (c *Clock) StartMonitoringClockJumps(
+	maxClockJumpCh <-chan time.Duration,
+	tickerFn func(d time.Duration) *time.Ticker,
+	tickCallback func(),
+) error {
+	alreadyMonitoring := c.setMonitoringClockJump()
+	if alreadyMonitoring {
+		return errors.New("clock jumps are already being monitored")
+	}
+
+	go func() {
+		ticker := tickerFn(c.maxOffset)
+		ticker.Stop()
+		for {
+			select {
+			case maxClockJump, ok := <-maxClockJumpCh:
+				ticker.Stop()
+				if !ok {
+					return
+				}
+
+				refreshLastUpdateTimeItvl := maxClockJump / 2
+				if refreshLastUpdateTimeItvl > 0 {
+					ticker = tickerFn(refreshLastUpdateTimeItvl)
+				}
+				c.updateMaxClockJump(maxClockJump)
+			case <-ticker.C:
+				c.PhysicalNow()
```

Comment:
what's the purpose of this?

Reply:
To detect clock jumps, we do `lastPhysicalTime - newPhysicalTime`. This cron makes sure that `lastPhysicalTime` is up to date. Otherwise, if no onw calls `c.PhysicalNow()` for a long period, it would be detected as a jump in the next call to `c.PhysicalNow()`  even though there was no jump